ast/compile.go

```diff
@@ -319,7 +320,30 @@ func (c *Compiler) WithBuiltins(builtins map[string]*Builtin) *Compiler {
 	return c
 }
 
-// WithUnsafeBuiltins will add all built-ins in the map to the "blacklist".
+// WithCapabilities sets capabilities to enable during compilation. Capabilities allow the caller
+// to specify the set of built-in functions available to the policy.
+//
+//  - If called BEFORE `WithBuiltins`, builtins added by `WithBuiltins` are added.
+//  - If called AFTER `WithBuiltins`, builtins added in WithBuiltins are dropped.
```

Comment:
This kind of ordering thing on the like `WithXX` stuff is new (or at least I don't think we do this other places). Would it make more sense to just hold the capabilities and apply them at compile time?

If someone was adding builtins _and_ using capabilities could they add the new ones to the capabilities before passing them in to the compiler?

Reply:
I've pushed another commit that (I think) improves this a bit. It relies on lazily initializing the compiler when `Compile()` is called. Going forward, we should be able to just deprecate the WithBuiltins function because callers can construct the Capabilities structure to include their custom built-in functions. Let me know what you think.